File: A/C/192A.c
#include <stdio.h>
#include <math.h>

int main()
{
  int n, i, k, r, x, found = 0;
  scanf("%d", &n);
  n *= 2;
  k = (sqrt(n * 1.0f) + 0.5f);
  for (i = 1; i < k; i ++)
  {
    r = n - i * i - i;
    x = (int)sqrt(r * 1.0f);
    if (x * (x + 1) == r)
    {
      found = 1;
      break;
    }
  } 
  if (found)
  {
    printf("YES");
  }
  else
  {
    printf("NO");
  }
  return (0);
}

File: A/CS/192A.cs
using System;

namespace ConsoleApplication2
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            n *= 2;
            int k = (int)Math.Sqrt(n);
            bool found = false;
            for (int i = 1; i < k; i++)
            {
                int r = n - i * i - i;
                int x = (int)Math.Sqrt(r);
                if (x * (x + 1) == r)
                {
                    found = true;
                    break;
                }
            }
            if (found)
            {
                Console.Write("YES");
            }
            else
            {
                Console.Write("NO");
            }
            //Console.ReadLine();
        }
    }
}
